packages/components/collapsible/src/collapsible.ts

```diff
@@ -0,0 +1,236 @@
+import type { ComponentPublicInstance, PropType, Ref } from 'vue'
+import { Transition, computed, defineComponent, h, ref, toRefs } from 'vue'
+import type { ElementType } from '@oku-ui/primitive'
+import type { Scope } from '@oku-ui/provide'
+import { createProvideScope } from '@oku-ui/provide'
+import { Primitive } from '@oku-ui/primitive'
+import { composeEventHandlers } from '@oku-ui/utils'
+
+import { useControllableRef, useId } from '@oku-ui/use-composable'
+
+// import { useId } from '../../../core/use-composable/src/index'
+
+// ---Collapsible---
+interface CollapsibleProps { }
+type CollapsibleElement = ElementType<'div'>
+type CollapsibleTriggerElement = ElementType<'button'>
+type CollapsibleContextValue = {
+  contentId: string
+  disabled?: Ref<boolean>
+  open: Ref<boolean>
+  onOpenToggle(): void
+}
+
+const COLLAPSIBLE_NAME = 'Collapsible'
+const [createCollapsibleContext, createCollapsibleScope] = createProvideScope(COLLAPSIBLE_NAME)
+const [collapsibleProvider, useCollapsibleContext]
+  = createCollapsibleContext<CollapsibleContextValue>(COLLAPSIBLE_NAME)
+
+const Collapsible = defineComponent({
+  name: COLLAPSIBLE_NAME,
+  inheritAttrs: false,
+  props: {
+    defaultOpen: {
+      type: Boolean,
+    },
+    open: {
+      type: Boolean,
+      default: false,
+    },
+    disable: {
+      type: Boolean,
+      default: false,
+    },
+    scopeCollapsible: {
+      type: Object as unknown as PropType<Scope>,
+      required: false,
+    },
+    onOpenChange: {
+      type: Function as PropType<(open: boolean) => void>,
+    },
+  },
+  setup(props, { attrs, slots, expose }) {
+    const innerRef = ref<ComponentPublicInstance>()
+    const { ...CollapsibleProps } = attrs as CollapsibleElement
+    const { disable, scopeCollapsible, open, defaultOpen } = toRefs(props)
+
+    // const { _ref: collapsibleRef, refEl: collapsibleRefEl } = useRef<CollapsibleElement>()
+
+    expose({
+      innerRef: computed(() => innerRef.value?.$el),
+    })
+
+    const { state } = useControllableRef({
+      prop: open.value,
+      onChange: props.onOpenChange,
+      defaultProp: defaultOpen.value,
+    })
+
+    collapsibleProvider({
+      contentId: useId(),
+      disabled: disable,
+      open: state,
+      onOpenToggle() {
+        // TODO: Need Help!!!
+        state.value = !state.value
```

Comment:
Need Help !!!

Reply:
I spent a long time trying to solve the issue of modifying data between parent and child components, but still haven't found a solution.

Reply:
I've fixed many builds and bugs. [check new commit.](https://github.com/oku-ui/primitives/pull/199/commits/838d891b073c63adbff1d625674080219018096a)

Reply:
Thank you! You're amazing. I'll learn from you.